api/src/main/java/org/apache/iceberg/transforms/Truncate.java

```diff
@@ -252,7 +254,7 @@ public CharSequence apply(CharSequence value) {
         return null;
       }
 
-      return UnicodeUtil.truncateString(value, length);
+      return UnicodeUtil.truncateStringUnsafe(length, value);
```

Comment:
This change keeps the same logic but avoids checking the `width` in a precondition for every string.

The width is already validated in the constructor and this is a hot code path.

Reply:
Why are the arguments transposed? I think the new method should use the same argument order that the other `UnicodeUtil` method does.

Reply:
The arguments were transposed to match the other ones from `TruncateUtil`. But I'll update it as it is somewhat odd to be transposed within `UnicodeUtil`.

Reply:
This wound up being removed anyway. In a separate PR, I'll update this `Transform` path to not validate the input width on every call but it's out of scope for this PR.

Reply:
I opened https://github.com/apache/iceberg/pull/5496 to skip validating the input `length` for CharSequence Truncate on every record.